packages/dhis2/src/Adaptor.js

```diff
@@ -397,26 +422,51 @@ export function update(
   return state => {
     console.log(`Preparing update operation...`);
 
+    let mappedOptions;
+
+    if (shouldUseNewTracker(resolvedResourceType)) {
+      mappedOptions = {
+        ...options,
+        importStrategy: 'UPDATE',
+      };
+    } else {
+      mappedOptions = options;
+    }
+
     const resolvedResourceType = expandReferences(resourceType)(state);
     const resolvedPath = expandReferences(path)(state);
     const resolvedData = expandReferences(data)(state);
-    const resolvedOptions = expandReferences(options)(state);
+    const resolvedOptions = expandReferences(mappedOptions)(state);
```

Comment:
Again, the order is wrong on the options resolution here

Reply:
Yeah same as https://github.com/OpenFn/adaptors/pull/819#discussion_r1838628253

Always resolve options first, before doing anything else